include/http/request_handler.js

```diff
@@ -534,15 +535,26 @@ module.exports = function RequestHandlerModule(pb) {
      */
     RequestHandler.prototype.handleRequest = function(){
 
-        //fist things first check for public resource
+        //first things first check for public resource
         if (RequestHandler.isPublicRoute(this.url.pathname)) {
             return this.servePublicContent();
         }
 
         //check for session cookie
         var cookies = RequestHandler.parseCookies(this.req);
+        var siteObj = RequestHandler.sites[this.hostname];
+        var url = this.url.pathname;
         this.req.headers[pb.SessionHandler.COOKIE_HEADER] = cookies;
 
+        if(typeof(siteObj) !== 'undefined' && siteObj.hasOwnProperty('forceLocale') && siteObj.forceLocale) {
+            cookies.locale = cookies.locale || cookies[" locale"];
+            cookies.locale = cookies.locale || siteObj.defaultLocale;
+            if (checkRouteContainsLocale(url, siteObj)) {
+                var redirectLocation = pb.SiteService.getHostWithProtocol(this.hostname);
+                redirectLocation += '/' + cookies.locale + this.url.path;
+                return this.doRedirect(redirectLocation, pb.HttpStatus.MOVED_TEMPORARILY);
+            }
+        }
```

Comment:
The ultimate goal of this code and our goal with this set is to force the locale into the route.  In the Multilingual work that has already been done, you added support for locale in the route if you specify the right prop int he route object.  (Localization: true)

That _allows_ the user to use / the same as /en-US or /fr-FR or ect.  Our use case was to say if the user types in: www.mysite.com, we need to redirect to www.mysite.com/en-US  We wanted to have a way to force the user to get the locale in the route, regardless of whether or not they typed it in.  

We did this by storing their locale in a cookie.  The code in the request handler was to read that cookie we set, and then redirect the user from / to /en-US or /fr-FR or whatever locale the cookie says they should be on. 

I would not be opposed to handling this in a plugin, if there was a way to hook into the request handler from a plugin.  I also would not mind if there was some way to build in this ability (to force the locale into the route) at the platform level.

I hope that makes more sense, if you have any further questions please let me know.